scipy/spatial/transform/_rotation.pyx

```diff
@@ -2766,33 +2806,65 @@ cdef class Rotation:
 
         where :math:`w_i`'s are the `weights` corresponding to each vector.
 
-        The rotation is estimated with Kabsch algorithm [1]_.
+        The rotation is estimated with Kabsch algorithm [1]_, and solves what
+        is known as the "pointing problem", or "Wahba's problem".
+
+        There are two special cases. The first is when one of the weights is
+        infinite. In this case, the rotation is estimated by first applying the
+        Kabsch algorithm to the subset of vectors with finite weights. The
+        rotation is then found from that result, to the infinite-weight vectors
```

Comment:
What does it mean exactly "rotation is then found from that result"?

Does it correspond to the same optimization problem with the constraint?

Reply:
I reworded this because yeah it was a bit confusing. Basically, the baseline Kabsch algorithm is run on all the non-infinite weights, and then the resulting vectors from that rotation matrix are taken as the single pair of non-infinite vectors for the align-constrain case. I added some tests in `test_align_vectors_near_inf` to confirm approximate equality of a set of 3 vectors with `weights=[1e6, 2, 1]` and `weights=[np.inf, 2, 1]`